main.cpp

```diff
@@ -15,7 +15,9 @@ int main(int argc, char* argv[])
   //close the file right away so if the program fails anywhere else, the log file won't be left open.
   //the log file is opened anywhere it needs to be read from or written to, so it is ok to close it here.
   std::ofstream ofile("request_response_log.txt");
```

Comment:
yeah just a safety kind of thing and a design decision just in case the things are called in a specific order